src/middlewared/middlewared/plugins/smb.py

```diff
@@ -1003,6 +1044,31 @@ async def do_delete(self, id):
 
         return result
 
+    @filterable
+    async def query(self, filters=None, options=None):
+        """
+        Query shares with filters. In clustered environments, local datastore query
+        is bypassed in favor of clustered registry.
+        """
+        extra = options.get('extra', {})
+        try:
```

Comment:
Shouldnt we make sure ha_mode exists? Letting a KeyError be raised on None almost every time seems bad